internal/service/network/bridge_driver.go

```diff
@@ -0,0 +1,375 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package network
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strconv"
+
+	"github.com/containerd/nerdctl/pkg/lockutil"
+	"github.com/containerd/nerdctl/pkg/netutil"
+	"github.com/coreos/go-iptables/iptables"
+	"github.com/runfinch/finch-daemon/api/types"
+	"github.com/runfinch/finch-daemon/internal/backend"
+	"github.com/runfinch/finch-daemon/pkg/flog"
+)
+
+const (
+	FinchICCLabel = "finch.network.bridge.enable_icc"
+
+	BridgeICCOption             = "com.docker.network.bridge.enable_icc"
+	BridgeHostBindingIpv4Option = "com.docker.network.bridge.host_binding_ipv4"
+	BridgeNameOption            = "com.docker.network.bridge.name"
+)
+
+//go:generate mockgen --destination=../../../mocks/mocks_network/bridge_driver.go -package=mocks_network -mock_names BridgeDriverOperations=BridgeDriver . BridgeDriverOperations
+type BridgeDriverOperations interface {
+	HandleCreateOptions(request types.NetworkCreateRequest, options netutil.CreateOptions) (netutil.CreateOptions, error)
+	HandlePostCreate(net *netutil.NetworkConfig) (string, error)
+	SetBridgeName(net *netutil.NetworkConfig, bridge string) error
+	GetBridgeName(net *netutil.NetworkConfig) (string, error)
+	GetNetworkByBridgeName(bridge string) (*netutil.NetworkConfig, error)
+	DisableICC(bridgeIface string, insert bool) error
+	SetICCDisabled()
+	ICCDisabled() bool
+}
+
+// IPTablesWrapper is an interface that wraps the methods of iptables.IPTables
+// to help with mock
+//
+//go:generate mockgen --destination=../../../mocks/mocks_network/iptables.go -package=mocks_network  . IPTablesWrapper
+type IPTablesWrapper interface {
+	ChainExists(table, chain string) (bool, error)
+	NewChain(table, chain string) error
+	Insert(table, chain string, pos int, rulespec ...string) error
+	Append(table, chain string, rulespec ...string) error
+	DeleteIfExists(table, chain string, rulespec ...string) error
+}
+
+// IPTablesWrapperImpl implements IPTablesWrapper
+// that delegates to an actual iptables.IPTables instance.
+type IPTablesWrapperImpl struct {
+	ipt *iptables.IPTables
+}
+
+func NewIPTablesWrapper() IPTablesWrapper {
+	iptables, _ := iptables.NewWithProtocol(iptables.ProtocolIPv4)
```

Comment:
should we also consider IPv6?

Reply:
I have updated the code to handle Ipv6 enabled requests. So when Ipv6 is enabled, we update both the ipv4 and ipv6 Iptables rules to add the drop rule. Finch does not have any way to disable Ipv4, just enable/disable ipv6. So, default behavior is to awlways add the rule to the ipv4 iptables. And conditionally add the rule to the ipv6 tables if it is enabled.